content/en/blog/kubesphere-porterlb-implement-grayscale-release-on-bare-metal-kubernetes.md

```diff
@@ -0,0 +1,117 @@
+     name: porter-layer2-eip
+   spec:
+     # Use an unoccupied address on the same network segment as your K8s cluster.
+     address: 192.168.0.100
+     interface: eth0
+     protocol: layer2
+   ```
+   
+6. Run the `kubectl apply -f eip.yaml` command to create the Eip object, which functions as an IP address pool for PorterLB.
+
+Step 2: Create an application.
+1. Choose **Application Workloads** on the left navigation bar and click **Create Composing Application** on the right.
+2. Set **Application Name** to **demo-app**, **Application Version(Optional)** to **v1**, **Application Governance** to **On**, and click **Next**.
+3. Click **Add Service**, click **Stateless Service**, set **Name** to `demo-svc`, and click **Next**.
+5. Click **Add Container Image**, set **Image** to `luksa/kubia`, **Container Port** to `8080`, **Service Port** to `80`, click **√**, and click **Next**.
+6. Click **Next** on the **Mount Volumes** tab, click **Add** on the **Advanced Settings** tab, and click **Next**.
+7. Click **Add Route Rule**, click **Specify Domain**, set **HostName** to `demo.com`, **Paths** to `/path | demo-svc | 80`, and click **OK**.
+8. Click **Create**.
+
+Step 3: Configure grayscale release.
+1. Choose **Grayscale Release** on the left navigation bar, move the cursor to **Canary Release**, and click **Create Job**.
+2. Set **Release Job Name** to `demo-canary` and click **Next**.
+3. Select `demo-app` from the drop-down list, click **Select** on the right of **demo-svc**, and click **Next**.
+4. Set **Grayscale Release Version Number** to `v2` and click **Next**.
+5. Click **Create** on the **Policy Config** tab.
+
+Step 4: Test grayscale release.
+
+1. Choose **Grayscale Release** on the left navigation bar, click the **Job Status** tab, and click **demo-canary** on the right. 
+
+2. In the **Real-time traffic distribution** area, move the slider so that 100% traffic is sent to v2.
+
+3. Log in to a client machine connected to the gateway Service IP address (configured in the Eip object) and add the domain name information to the `etc/hosts` file:
+
+   ``` bash
+   192.168.0.100 demo.com
+   ```
+
+4. On the client machine, run the `curl demo.com/path` command for multiple times to access the application Service.
+
+   If grayscale release functions properly, only v2 can be accessed.
+
+   <img src="/images/blog/kubesphere-porterlb-implement-grayscale-release-on-bare-metal-kubernetes/v2.png" width="600px">
+
+5. On KubeSphere, repeat step 2 so that 100% traffic is sent to v1.
+
+6. On the client machine, run the `curl demo.com/path` command for multiple times to access the application Service.
+
+   If grayscale release functions properly, only v1 can be accessed.
+
+   <img src="/images/blog/kubesphere-porterlb-implement-grayscale-release-on-bare-metal-kubernetes/v1.png" width="600px">
+
+7. After v2 is fully tested, you can set v2 to take over all traffic and take the canary release job offline to form1ally release v2.
```

Comment:
If we set the slider to 50%, will the grayscale release introduces 50% of traffic to v1 and v2 respectively?

Reply:
Yes. That is how grayscale release is meant to work.